### ISSUE TITLE
Import middlewares

### ISSUE THREAD

**Issue by JeanCigoli** [tulios/mappersmith#403]
Hello everything is fine?

We are using this lib in our project, and on Friday we started having import errors.

This error here:

`Package subpath './middlewares/encode-json' is not defined by \"exports\" in /code/node_modules/mappersmith/package.json`

I would like to know if the new version will not have compatibility with this import, and if we will have to change how we do it?

(Sorry for my English, but I'm Brazilian and I used Google Translate to write)

**Comment by klippx** [comment]
It sounds like a legit regression, and thank you for reporting. 

Explanation: In 2.43.0 we changed how the `dist/` folder was built, we also changed the `package.json` exports fields for the npm package, and I tried my hardest to ensure backwards compatibility. But we broke you, it seems.

Can you please share how you are importing which is no longer working?

**Comment by roodie-ree** [comment]
For us it fails at this line:
```
const EncodeJson = require('mappersmith/middlewares/encode-json').default
```

Changing it to 
```
const EncodeJson = require('mappersmith/middleware/encode-json').default
```
seems to fix it, so it seems the re-export inside `middlewares` is broken.

**Comment by klippx** [comment]
We must have missed the case were people imported from this folder.

Thanks, I have added a regression test to ensure this wont happen again.

**Comment by klippx** [comment]
This should be fixed in 2.43.1, please give it a try!
